conf.d/pure.fish

```diff
@@ -60,11 +60,14 @@ _pure_set_default pure_color_right_prompt $pure_color_normal
 
 # VI mode indicator
 # true (default):  indicate a non-insert mode by reversing the prompt symbol (❮)
-# false:           indicate vi mode with [I], [N], [V]  
+# false:           indicate vi mode with [I], [N], [V]
 _pure_set_default pure_reverse_prompt_symbol_in_vimode true
 
 # Title
 _pure_set_default pure_symbol_title_bar_separator "—"
 
 # Whether or not is a fresh session
 set --global _pure_fresh_session true
+
+# Register `_pure_prompt_new_line` as an event handler fot `fish_prompt`
+functions -q _pure_prompt_new_line
```

Comment:
Why do you register `_pure_prompt_new_line`, doesn't the [autoload mechanism](https://fishshell.com/docs/current/tutorial.html#tut_autoload) already do this?

Otherwise, I think we should limit _conf.d/pure.fish_ to public variables and extract `_pure_prompt_new_line` and `_pure_fresh_session` to another file (_e.g._ _conf.d/pure_event_handler.fish_).
If need be, source said file in _conf.d/pure.fish_, but that would allow us to separate public configuration from internal plumbing.

Reply:
> ...doesn't the autoload mechanism already do this?

Autoload mechanism only loads functions on their invocations, but in this case we need to register function as event handler. So if we won't source `_pure_prompt_new_line.fish` it's never gonna be registered and called.

When we query function with `functions -q` we actually use autoloading mechanism to source `_pure_prompt_new_line.fish` without need to specify filename and path manually. I think it's cleaner than using `source` directly.

Reply:
My bad I thought we were calling `_pure_prompt_new_line` at some point.

So do you we could still extract handler registration and `_pure_fresh_session` to a dedicated file?

Reply:
@edouard-lopez on your second point. I want to extract all implementation details (something that if changed would brake `pure`) to separate file:

```
# Deactivate the default virtualenv prompt so that we can add our own
set --global --export VIRTUAL_ENV_DISABLE_PROMPT 1

# Whether or not is a fresh session
set --global _pure_fresh_session true

# Register `_pure_prompt_new_line` as an event handler fot `fish_prompt`
functions -q _pure_prompt_new_line
```

Let's decide how we should call it. `_pure_variables.fish` perhaps?

Reply:
`_pure_internals.fish`?

Reply:
I'd say `_` in `_pure` already signifies it's private, so I would prefer `_pure_init.fish`, but that's not a strong opinion